// typescript-axios-client-generated 2/models/booking-dto.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Student Tutor Booking Management System
 * This is a system allowing students and tutors to book in-person sessions at designated venues on campus.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: 222641495@mycput.ac.za
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

 /**
 * 
 *
 * @export
 * @interface BookingDTO
 */
export interface BookingDTO {

    /**
     * @type {number}
     * @memberof BookingDTO
     */
    bookingId?: number;

    /**
     * @type {number}
     * @memberof BookingDTO
     */
    tutorId: number;

    /**
     * @type {number}
     * @memberof BookingDTO
     */
    studentId: number;

    /**
     * @type {number}
     * @memberof BookingDTO
     */
    subjectId: number;

    /**
     * @type {string}
     * @memberof BookingDTO
     */
    date: string;

    /**
     * @type {string}
     * @memberof BookingDTO
     */
    startTime: string;

    /**
     * @type {string}
     * @memberof BookingDTO
     */
    endTime: string;

    /**
     * @type {number}
     * @memberof BookingDTO
     */
    locationId: number;

    /**
     * @type {string}
     * @memberof BookingDTO
     */
    topic: string;
}
