// typescript-axios-client-generated 2/models/tutor-dto.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Student Tutor Booking Management System
 * This is a system allowing students and tutors to book in-person sessions at designated venues on campus.
 *
 * OpenAPI spec version: 1.0.0
 * Contact: 222641495@mycput.ac.za
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 * Do not edit the class manually.
 */

 /**
 * 
 *
 * @export
 * @interface TutorDTO
 */
export interface TutorDTO {

    /**
     * @type {number}
     * @memberof TutorDTO
     */
    tutorId?: number;

    /**
     * @type {string}
     * @memberof TutorDTO
     */
    name: string;

    /**
     * @type {string}
     * @memberof TutorDTO
     */
    lastName: string;

    /**
     * @type {string}
     * @memberof TutorDTO
     */
    email: string;

    /**
     * @type {string}
     * @memberof TutorDTO
     */
    phoneNumber: string;

    /**
     * @type {string}
     * @memberof TutorDTO
     */
    password: string;
}
